### ISSUE TITLE
Picker crashes immediately on 5.6.0 in RDP sessions (0xc0000005, regression from 5.5.5)

### ISSUE THREAD

**Issue by mukkino** [aloneguid/bt#203]
- bt.exe 5.6.0 picker crashes with exception 0xc0000005 at fault offset 0x0000000000094243
- Every single picker invocation crashes at the same offset
- Regression from 5.5.5 which works correctly
- Running via RDP session with AMD Radeon graphics + Microsoft Remote Display Adapter

**Comment by mukkino** [comment]
## 1. Reproduction steps

1. Install BT 5.6.0 via `winget` on a session using **Microsoft Remote Display Adapter (RDP)** + **AMD Radeon**
2. Set `always = y` in `[picker]`
3. Click any link
4. Picker does not appear

## 2. Crash evidence from Windows Event Log

- **Faulting application:** `bt.exe`, version `5.6.0.0`
- **Faulting module:** `bt.exe` (self — not a DLL)
- **Exception code:** `0xc0000005` (access violation / null dereference)
- **Fault offset:** `0x0000000000094243`

This offset is identical on every single crash — **100% deterministic**, not flaky.

## 3. What the log tells us

```text
[timestamp] showing rule picker for [url]
[timestamp] started with args: [url]|[hwnd]
```

The picker process starts, logs one line, then crashes. No window ever appears. Our Win32 `EnumWindows` scan confirmed no picker window exists at any point.

## 4. Root cause hypothesis (with source code reference)

From `picker_app.cpp` line 20:

```cpp
app{grey::app::make(title, 100, 100)},
```

Immediately followed by:

```cpp
app->win32_can_resize = false;   // crash here if make() returned null
app->win32_center_on_screen = true;
```

If `grey::app::make()` returns `null` (or an empty `unique_ptr`) on this display configuration, the first dereference produces a deterministic access violation. The `config_app` (main BT window) works fine — same grey framework — so something changed specifically in how `make()` initialises for the picker context in 5.6.0.

## 5. Confirmation it's a regression

- BT `5.5.5` works perfectly on the same machine, same session, same display adapter
- Immediate rollback to `5.5.5` via `winget install --version 5.5.5` resolves it

## 6. Display/GPU config (already ruled out as root cause)

- **Microsoft Remote Display Adapter** (RDP session) + **AMD Radeon 31.0.21924.61** (Dec 2025 driver)
- **Parsec Virtual Display Driver** also tested — same crash at same offset

Therefore it's not a display adapter issue

## 7. What makes no difference

- `show_native_chrome = y` or `n` — same crash either way
- `close_on_focus_loss = n` — same crash
- Icon paths in `config.ini` — same crash

**Comment by aloneguid** [comment]
whcih windows version are you using?

**Comment by mukkino** [comment]
OsName = Microsoft Windows 11 Pro
OSDisplayVersion = 25H2
OsBuildNumber = 26200
UBR = 8037
Windows 11 Pro 25H2, build 26200.8037

**Comment by aloneguid** [comment]
I can't reproduce this in Windows sandbox but will keep looking.

**Comment by mukkino** [comment]
Thank you

**Comment by aloneguid** [comment]
Do you mind trying 5.6.2 @mukkino ?

**Comment by mukkino** [comment]
5.6.2 and 5.6.3 not working. Latest working 5.5.5

Faulting application name: bt.exe, version: 0.0.0.0, time stamp: 0x69dd6d5d
Faulting module name: bt.exe, version: 0.0.0.0, time stamp: 0x69dd6d5d
Exception code: 0xc0000005
Fault offset: 0x0000000000092ab3
Faulting process id: 0x95D8
Faulting application start time: 0x1DCD054E72D4EE8

**Comment by aloneguid** [comment]
@mukkino it's unclear why the crash happens, but most likely this is due to third party software interference. If you are happy to capture a dump and send it over you can follow these instructions:

- Download [ProcDump](https://learn.microsoft.com/en-us/sysinternals/downloads/procdump).
- Before launching BT, run the following command in the terminal: `procdump -mm -w -e bt`.
- Launch bt and make it crash.
- ProcDump will say something like "[18:47:44]Dump 1 initiated: C:\Users...\bt.exe_260202_184744.dmp".
- Please zip and send the .dmp file to aeov7d47e@mozmail.com (temporary email).
That will pin point exactly where the crash is happening in code.

**Comment by mukkino** [comment]
I captured a crash dump from 5.6.3 and sent it to that email address. Sorry that it has been sanitized a bit, but the relevant information should still be intact.

This may also be useful for debugging: the first command you gave me did not work.

ProcDump showed:

Exception monitor: Unhandled

and then:

Process Exit: PID 54204, Exit Code 0xc0000005
Dump count not reached

So it seems ProcDump was waiting for an unhandled exception trigger with `-e`, but the process exited before it could catch a matching exception event. Because of that, I created the dump with:

`procdump -mm -t -w bt`

What I can confirm from the dump:
- Exception: 0xc0000005 (access violation)
- Module: bt.exe
- Fault offset: 0x926b3
- Access type: read
- Address being read: 0x8
- At the crash point, RAX is null

This makes it look like a null-pointer dereference inside `bt.exe` itself rather than random corruption.

**Comment by aloneguid** [comment]
Thanks for this. Unfortunately crash does not point anywhere inside bt.exe (procdump not catching it also is a red flag). I am wondering if you have any third-party software interfering i.e. Windhawk etc. Or it's reproducible elsewhere?

**Comment by aloneguid** [comment]
@mukkino  Can you try to start clean (rename `%localappdata%/bt` folder to something like `bt.backup`) and see if the crash still happening Please.

**Comment by Michi05** [comment]
hi all,

I'd like to help but not sure how.
Version 5.5.5 works as a charm (and I love it), but I did find this issue before downgrading, while running locally (not RDP) on my laptop with the following specs:

>   OS:       Microsoft Windows 11 Home 64-bit
>   Build:    10.0.26200 (Build 26200)

>   CPU:      Intel Core Ultra 7 258V
>   Cores:    8 cores / 8 threads
>   Speed:    2200 MHz max (Lunar Lake, Intel 4 process node)
> 
>   RAM:      32 GB
> 
>   GPU:      Intel Arc 140V GPU — 16 GB shared (LPDDR5X)
>   Driver:   32.0.101.8132
>   (+ Meta Virtual Monitor and SuperDisplay Virtual Adapter also present)
> 
> NOTE: GPU integrated & sharing the 32 GB LPDDR5X memory pool.

And I got the following from procdump with the version installed from _winget_:

> > _procdump -mm -t -w bt.exe_
> 
> ProcDump v11.1 - Sysinternals process dump utility
> Copyright (C) 2009-2025 Mark Russinovich and Andrew Richards
> Sysinternals - www.sysinternals.com
> 
> Waiting for process named bt.exe...
> 
> Process:               bt.exe (43188)
> Process image:         C:\Users\user\AppData\Local\Microsoft\WinGet\Packages\aloneguid.bt_Microsoft.Winget.Source_8wekyb3d8bbwe\bt.exe
> CPU threshold:         n/a
> Performance counter:   n/a
> Commit threshold:      n/a
> Threshold seconds:     n/a
> Hung window check:     Disabled
> Log debug strings:     Disabled
> Exception monitor:     Disabled
> Exception filter:      [Includes]
>                        *
>                        [Excludes]
> Terminate monitor:     Enabled
> Cloning type:          Disabled
> Concurrent limit:      n/a
> Avoid outage:          n/a
> Number of dumps:       1
> Dump folder:           C:\Users\user\AppData\Local\Microsoft\WinGet\Packages\aloneguid.bt_Microsoft.Winget.Source_8wekyb3d8bbwe\
> Dump filename/mask:    PROCESSNAME_YYMMDD_HHMMSS
> Queue to WER:          Disabled
> Kill after dump:       Disabled
> 
> 
> Press Ctrl-C to end monitoring without terminating the process.
> 
> [19:41:23]Process Exit: PID 43188, Exit Code 0xc0000005
> [19:41:23]Dump 1 info: Available space: 499525033984
> [19:41:23]Dump 1 initiated: C:\Users\user\AppData\Local\Microsoft\WinGet\Packages\aloneguid.bt_Microsoft.Winget.Source_8wekyb3d8bbwe\bt.exe_260428_194123.dmp
> [19:41:23]Dump 1 complete: 13 MB written in 0.2 seconds
> [19:41:23]The process has exited.
> [19:41:23]Dump count reached.

Weird quirk is that it followed the 'instagram > browser" rule properly when I tried it this time, which I hadn't noticed before, so wondering if it only happens when loading the UI on fallback? I'll think of some tests to prove this if it helps

Thanks for your work!
Miguel

**Comment by aloneguid** [comment]
Hi, could you try 5.6.9 please.

**Comment by Michi05** [comment]
> Hi, could you try 5.6.9 please.

Hello Ivan,
It seems to be working perfectly for me now.
Thank you so much for your work and for taking care of this so quickly!


(Img just as checkpoint)
<img width="1062" height="218" alt="Image" src="https://github.com/user-attachments/assets/73dfbc61-a034-4ec9-9ec0-7d50b63b50bc" />

**Comment by aloneguid** [comment]
You're welcome @Michi05, and thanks for the beverage, tastes good

**Comment by mukkino** [comment]
It works now for me as well! Amazing! Thank you
